### ISSUE TITLE
Deleting any child entity of the Atom Default Environment causes a warning to be printed to the Console

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#4863]
**Describe the bug**
Deleting any of the Atom Default Environment children entities (Grid, Shader Ball, Global Sky, Camera, Sun or Ground) causes an error to be printed to the Console:
`[Warning] (Prefabs) - Prefab - EntityIdMapper: Entity with Id [<Entity ID>] has no registered owning instance`.

It should be noted that this does not occur when deleting the Atom Default Environment parent entity itself and any newly created entities.

**Steps to reproduce**
1. Create a new level.
2. Delete any of the children entities under the Atom Default Environment entity.

**Expected behavior**
No warning is printed to the Console.

**Actual behavior**
A warning is printed to the Console.

**Video**

https://user-images.githubusercontent.com/86952082/138267197-7f8181f4-6410-44b6-bcfb-42e3da52de33.mp4

**Found in Branch**
Development (74d7405)

**Desktop**
Device: PC
OS: Windows
Version 10
CPU AMD Ryzen 5 3600 6-Core Processor, 3600 MHz, 6 Core(s), 12 Logical Processor(s)
GPU NVIDIA GeForce RTX 2060 SUPER
Memory 16GB

**Comment by LB-ArturSiewniak** [comment]
On Stabilization/2111RTE (ac03b6f) build another reproduction steps were found to print these warnings. Assigning or modifying an assigned Script Canvas graph which contains an empty EntityID variable with its Initial Value Source set to From Component causes it as well. Should it be logged as a separate issue? Please refer to the attached video for more information:

https://user-images.githubusercontent.com/86952082/145597895-f408e343-a7b1-4aa5-8126-82d368fab672.mp4

**Comment by monroegm** [comment]
Now produces the following: 
```
[Warning] (Prefab Serialization) - The target path for "remove" operation has an invalid index.
    Reason: The operation has halted during a merge operation for '/4' because a field or element has an invalid value.
[Error] (Prefab) - Some of the patches were not successfully applied.
```

**Comment by monroegm** [comment]
Similar to #2921

**Comment by srikappa-amzn** [comment]
This is fixed by : https://github.com/o3de/o3de/pull/8146

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (ba99568) build.
